clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1256,7 +1256,17 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
                         log.debug("Discarding stale fetch response for partition {} since the fetched offset {} " +
                                 "does not match the current offset {}", tp, fetchOffset, subscriptions.position(tp));
                     } else if (subscriptions.hasDefaultOffsetResetPolicy()) {
-                        log.info("Fetch offset {} is out of range for partition {}, resetting offset", fetchOffset, tp);
+                        if (partition.logStartOffset > 0) {
+                            if (isolationLevel == IsolationLevel.READ_COMMITTED) {
+                                log.info("Fetch offset {} is out of range for partition {}. We only have log segments in " +
+                                                "the range from {} to {}. Resetting offset",
+                                        fetchOffset, tp, partition.logStartOffset, partition.lastStableOffset);
+                            } else {
+                                log.info("Fetch offset {} is out of range for partition {}. We only have log segments in " +
+                                                "the range from {} to {}. Resetting offset",
+                                        fetchOffset, tp, partition.logStartOffset, partition.highWatermark);
+                            }
```

Comment:
Most of if-else is same code. How about this:

```
log.info("Fetch offset {} is out of range for partition {}. We only have log segments in " +
                                                "the range from {} to {}. Resetting offset",
                                        fetchOffset, tp, partition.logStartOffset,
isolationLevel == IsolationLevel.READ_COMMITTED ? partition.lastStableOffset : partition.highWatermark);
```

Reply:
Refactored using conditional operator